src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/SocketInput.cs

```diff
@@ -236,7 +250,7 @@ public void OnCompleted(Action continuation)
 
             if (ReferenceEquals(awaitableState, _awaitableIsNotCompleted))
             {
-                return;
+                // nop;
```

Comment:
Change `else` to `else if (!ReferenceEquals(awaitableState, _awaitableIsNotCompleted)`?